website/blog/2021-09-05-2.4.0.md

```diff
@@ -0,0 +1,248 @@
+<div
+  id="foo-bar-baz"
+  class="bar-foo-baz"
+  title="a sample title"
+  data-foo="bar"
+  data-bar="baz">
+  lorem ipsum dolor sit amet
+</div>
+```
+
+## Other Changes
+
+### JavaScript
+
+#### Support parenthesized tagged template literals for styled components ([#11246](https://github.com/prettier/prettier/pull/11246) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```js
+// Input
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position:   fixed;
+  color: red;
+  `;
+
+// Prettier 2.4
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position:   fixed;
+  color: red;
+  `;
+
+// Prettier 2.3
+const StyledComponent =
+  /** @type {import('styled-components').ThemedStyledFunction<'div',null,{overlap: boolean}>} */
+  (styled.div)`
+    position: fixed;
+    color: red;
+  `;
+
+```
+
+#### Count correctly regex literals in method chain arguments ([#11299](https://github.com/prettier/prettier/pull/11299) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```js
+// Input
+foo1(/𠮟𠮟𠮟/).foo2(bar).foo3(baz);
+
+foo1(/叱叱叱/).foo2(bar).foo3(baz);
+
+// Prettier 2.3
+foo1(/𠮟𠮟𠮟/)
+  .foo2(bar)
+  .foo3(baz);
+
+foo1(/叱叱叱/).foo2(bar).foo3(baz);
+
+// Prettier 2.4
+foo1(/𠮟𠮟𠮟/)
+  .foo2(bar)
+  .foo3(baz);
+
+foo1(/叱叱叱/)
+  .foo2(bar)
+  .foo3(baz);
+
+```
+
+#### Support hack-style pipeline proposal ([#11335](https://github.com/prettier/prettier/pull/11335) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+Support new [hack-style pipeline proposal](https://github.com/js-choi/proposal-hack-pipes). We chose `%` as the topic token, following the official explainer.
+
+Also, [smart-style pipeline proposal](https://github.com/js-choi/proposal-smart-pipelines) is removed. See https://prettier.io/docs/en/rationale.html#disclaimer-about-non-standard-syntax for the rationale behind this decision.
+
+```js
+// Input
+const foo = fn() |> fn1(%) |> fn2(%);
+
+// Prettier 2.3
+SyntaxError: Unexpected token (1:25)
+> 1 | const foo = fn() |> fn1(%) |> fn2(%);
+    |                         ^
+  2 |
+
+// Prettier 2.4
+const foo = fn() |> fn1(%) |> fn(%);
+
+```
+
+### TypeScript
+
+#### Fix formatting intersection types that includes mapped types ([#11247](https://github.com/prettier/prettier/pull/11247) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
+<!-- prettier-ignore -->
+```ts
+// Input
+type Example = {
+  [A in B]: T;
+} & {
+  [A in B]: T;
+};
+
+// Prettier 2.3
+type Example = {
+  [A in B]: T;
+} &
+  {
+    [A in B]: T;
+  };
+
+// Prettier 2.4
+type Example = {
+  [A in B]: T;
+} & {
+  [A in B]: T;
+};
+
+```
+
+### Flow
+
+#### Print leading semi before Flow variance sigils ([#11398](https://github.com/prettier/prettier/pull/11398) by [@noppa](https://github.com/noppa))
```

Comment:
```suggestion
#### Always put a semicolon before Flow variance sigils ([#11398](https://github.com/prettier/prettier/pull/11398) by [@noppa](https://github.com/noppa))
```

I don’t use Flow but is there a more common name for this syntax than “variance sigils?”  That name sounds a little obscure for everyday usage.

Reply:
I don't use Flow too so I'm not sure if it's commonly used. But "variance sigils" is used in Flow Docs (ref: https://flow.org/en/docs/types/generics/#toc-variance-sigils)